src/IslandoraContextManager.php

```diff
@@ -22,7 +30,9 @@ class IslandoraContextManager extends ContextManager {
   public function evaluateContexts(array $provided = []) {
 
     $this->activeContexts = [];
-
+    // XXX: Ensure that no earlier executed contexts in the request are still
+    // present.
+    $this->resetContextEvaluation();
```

Comment:
Just figured I'd get this written down, but... having thoughts that this might have something of a performance impact on typical site execution stuff, where contexts might be used for it's typical use case of block placements and influencing viewing and the like, wondering if it might make sense to only reset if `$provided` is _not_ empty, which I believe should only be when we are responding to something from `IslandoraUtils` (or some other mechanism that has _provided_ the contexts that we are to use).

Alternatively, might it make sense to roll our own  particular `islandora.context.manager` service or something of the like instead of [replacing the `context.manager` as we do presently](https://github.com/Islandora/islandora/blob/db859227658b4e218499f06b45e6230a850180ab/src/IslandoraServiceProvider.php#L13-L19), to avoid impacts on the more expected uses of the `context.manager` service?

Reply:
Met with @adam-vessey internally, going to attempt to implement his above suggestion.

Reply:
On second thought there is potentially some concern about BC breaking here if we were going to alter the `islandora.context.manager` in this way. Going to insulate the above resets for the time being in this commit and bring up the discussion at the next tech call.